google/cloud/aiplatform/models.py

```diff
@@ -2036,3 +2058,138 @@ def list(
+                set when the ``exportableContent`` field of the
+                [Model.supported_export_formats] object contains
+                ``ARTIFACT``.
+            image_destination (str):
+                The Google Container Registry or Artifact Registry URI where
+                the Model container image will be copied to. Accepted forms:
+
+                -  Google Container Registry path. For example:
+                ``gcr.io/projectId/imageName:tag``.
+
+                -  Artifact Registry path. For example:
+                ``us-central1-docker.pkg.dev/projectId/repoName/imageName:tag``.
+
+                This field
+                should only be set when the ``exportableContent`` field of
+                the [Model.supported_export_formats] object contains
+                ``IMAGE``.
+            sync (bool):
+                Whether to execute this deletion synchronously. If False, this method
+                will be executed in concurrent Future and any downstream object will
+                be immediately returned and synced when the Future has completed.
+
+        Raises:
+            ValueError if model does not support exporting.
+
+            ValueError if invalid arguments or export formats are provided.
+        """
+
+        # Model does not support exporting
+        if not self.supported_export_formats:
+            raise ValueError(f"The model `{self.resource_name}` is not exportable.")
+
+        # No destination provided
+        if not any((artifact_destination, image_destination)):
+            raise ValueError(
+                "Please provide an `artifact_destination` or `image_destination`."
+            )
+
+        export_format_id = export_format_id.lower()
+
+        # Unsupported export type
+        if export_format_id not in self.supported_export_formats:
+            raise ValueError(
+                f"'{export_format_id}' is not a supported export format for this model. "
+                f"Choose one of the following: {self.supported_export_formats}"
+            )
+
+        content_types = gca_model_compat.Model.ExportFormat.ExportableContent
+        supported_content_types = self._supported_export_formats[export_format_id]
+
+        if (
+            artifact_destination
+            and content_types.ARTIFACT not in supported_content_types
+        ):
+            raise ValueError(
+                "This model can not be exported as an artifact. Try exporting as "
+                "a container image by passing the `image_destination` argument."
+            )
+
+        if image_destination and content_types.IMAGE not in supported_content_types:
+            raise ValueError(
+                "This model can not be exported as a container image. Try exporting "
+                "the model artifacts by passing a `artifact_destination` argument."
+            )
+
+        # Construct request payload
+        output_config = gca_model_service_v1.ExportModelRequest.OutputConfig(
```

Comment:
Preference to use compat instead of v1 directly for these protos.

Reply:
This is being imported from `aiplatform.compat.types`, [see here](https://github.com/googleapis/python-aiplatform/pull/353/files#diff-72d6e2818d2224db618771b26094747754d49a2acf33160eaf017ff1b6ddf840R42). Is this not the correct way to access compat types?

Reply:
We should try to import from here to pin to the default version unless there is a reason to control the version in the implementation: https://github.com/googleapis/python-aiplatform/blob/master/google/cloud/aiplatform/compat/__init__.py

Reply:
Updating
```py
from google.cloud.aiplatform.compat.types import model_service_v1
```
to
```py
from google.cloud.aiplatform.compat.types import model_service
```